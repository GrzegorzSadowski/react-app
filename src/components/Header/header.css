.header {
  grid-row: 1/2;
  grid-column: 1/3;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.header--button {
  font-size: 1rem;
  background-color: #555555;
  padding: 5px 10px 5px 10px;
  margin-right: 1.5rem;
  border: none;
  color: #f65261;
  border-radius: 5px;
}
.activeResult--button {
  font-size: 2rem;
  background-color: transparent;
  padding: 5px 10px 5px 10px;
  margin-right: 1.5rem;
  border: none;
  color: #f65261;
  border-radius: 5px;
}
