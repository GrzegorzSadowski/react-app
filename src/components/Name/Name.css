h2 {
  margin: 0px;
  color: #f65261;
  padding-left: 1.5rem;
  font-size: 1.5rem;
}
