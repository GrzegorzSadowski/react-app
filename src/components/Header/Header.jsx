import "./header.css";
import React from "react";
import Name from "../Name/Name";
import Modal from "../Modal/Modal";
import ReactDOM from "react-dom";

class Header extends React.Component {
  state = {
    showModal: false,
  };

  handleOpenModalClick = () => {
    this.setState({ ...this.state, showModal: true });
  };

  handleCloseModalClick = () => {
    this.setState({ ...this.state, showModal: false });
  };

  render() {
    const { showButton } = this.props;

    return (
      <header className="header">
        <Name />
        {showButton && (
          <button
            onClick={this.handleOpenModalClick}
            className="header--button"
          >
            + ADD MOVIE
          </button>
        )}

        {this.state.showModal &&
          ReactDOM.createPortal(
            <Modal
              handleCloseModalClick={this.handleCloseModalClick}
              showAddMovie={true}
              showEditMovie={false}
              showDeleteMovie={false}
            />,
            document.getElementById("findYourMovie--container")
          )}
      </header>
    );
  }
}

export default Header;
