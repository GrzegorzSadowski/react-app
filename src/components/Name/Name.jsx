import './Name.css';
import React from 'react';


function Name() {
    return (
        <h2>netfilxroulette</h2>
    );
  }
  
  export default Name;