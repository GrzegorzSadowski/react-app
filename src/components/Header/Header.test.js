import React from "react";
import renderer from "react-test-renderer"
import Header from "./Header";
import Name from "../Name/Name";
import {shallow, configure} from "enzyme"
import Adapter from "@cfaester/enzyme-adapter-react-18"

test("Snapshot test when showButton is false", () =>{
    const tree = renderer.create(<Header showButton={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
})


test("Snapshot test when showButton is true", () =>{
    const tree = renderer.create(<Header showButton={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

configure({adapter: new Adapter()});


describe("rendrening components", () => {


it("show modal window after ADD MOVIE button is clicked", () =>
{
const component = shallow(<Header showButton={true}/>);
const button = component.find(".header--button");
expect(component.find("AddMovie").exists()).toBeFalsy;
button.simulate("click");
expect(component.find("AddMovie").exists()).toBeTruthy;
});

it("renders Header component Name withput crashing", () =>
{
const wrapper = shallow(<Header/>);
expect(wrapper.contains(<Name/>)).toEqual(true);
});
})

describe("passing props", ()=>{
it("show button ADD MOVIE if showButton is true", () =>
{
const component = shallow(<Header showButton={true}/>);
expect(component.find(".header--button").exists()).toBeTruthy;
});

})