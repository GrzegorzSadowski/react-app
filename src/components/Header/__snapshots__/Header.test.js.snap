// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test when showButton is false 1`] = `
<header
  className="header"
>
  <h2>
    netfilxroulette
  </h2>
</header>
`;

exports[`Snapshot test when showButton is true 1`] = `
<header
  className="header"
>
  <h2>
    netfilxroulette
  </h2>
  <button
    className="header--button"
    onClick={[Function]}
  >
    + ADD MOVIE
  </button>
</header>
`;
